cub/cub/util_policy_wrapper_t.cuh

```diff
@@ -0,0 +1,55 @@
+/******************************************************************************
+ * Copyright (c) 2011, Duane Merrill.  All rights reserved.
+ * Copyright (c) 2011-2020, NVIDIA CORPORATION.  All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *     * Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in the
+ *       documentation and/or other materials provided with the distribution.
+ *     * Neither the name of the NVIDIA CORPORATION nor the
+ *       names of its contributors may be used to endorse or promote products
+ *       derived from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL NVIDIA CORPORATION BE LIABLE FOR ANY
+ * DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ ******************************************************************************/
+
+#pragma once
+
+#include <cub/config.cuh>
+
+CUB_NAMESPACE_BEGIN
+
+#ifndef _CCCL_DOXYGEN_INVOKED // Do not document
+
+namespace detail
+{
+/**
+ * @brief Helper class template that allows overwriting the `BLOCK_THREAD` and `ITEMS_PER_THREAD`
+ * configurations of a given policy.
+ */
+// TODO(bgruber): this should be called something like "override_policy"
+template <typename PolicyT, int BLOCK_THREADS_, int ITEMS_PER_THREAD_ = PolicyT::ITEMS_PER_THREAD>
+struct policy_wrapper_t : PolicyT
+{
```

Comment:
Q: Why does this struct need to be moved into a separate file?

Reply:
Yeah, this is due to NVRTC. Specifically, `merge_sort.cuh` needs `policy_wrapper_t` but I can't include all of `util_device.cuh` due to NVRTC errors which I could not resolve easily, so my strategy with this and similar situations was to move things to separate headers. For the most part, I tried to fix the NVRTC errors but when that failed I resorted to this approach.

Reply:
I played a bit with NVRTC, and getting `util_device.cuh` to compile is surprisingly easy (you just `#ifdef` out everything from the header except `policy_wrapper_t`). So I guess pulling out that part makes a lot of sense!

I also tried getting the merge sort headers to compile, but I got stuck at thrust iterators, as always. This is what I meant today in the team meeting that we need an overhaul here :)

Reply:
Oh I wasn't aware I could do that. Would you like me to implement this instead of what I have now?

Reply:
No, just pull out `policy_wrapper_t`, it's fine!